python/deltalake/writer.py

```diff
@@ -195,7 +204,12 @@ def write_deltalake(
         partition_by = [partition_by]
 
     if table:  # already exists
-        if schema != table.schema().to_pyarrow(as_large_types=large_dtypes) and not (
+        hash_table_schema = _hash_schema(
+            table.schema().to_pyarrow(as_large_types=large_dtypes)
+        )
+        hash_schema_provided = _hash_schema(schema)
+
+        if hash_schema_provided != hash_table_schema and not (
```

Comment:
I think we should compare actual objects and not just hashes. It is still possible for hashes to collide, despite that being unlikely.

You could write a method to normalize schema order:

```python
def normalize_schema_order(schema: pa.Schema) -> pa.Schema:
    fields = sorted(iter(schema), key=lambda field: field.name)
    return pa.schema(fields)
```

And then compare those. How does that sound?